fzffm

```diff
@@ -141,15 +153,15 @@ done
 if [ "$(id -u)" -eq 0 ]; then
     # super user privilege
     yprompt () { # Usage yprompt [MSG] [BG_COLOR]
-	${FM_PROG} ${FM_OPT_REVERSE} ${FM_OPT_PROMPT} "SUDO: $1 "
+	    ${FM_PROG} $(makePromptOpts reverse prompt) "SUDO: $1 "
     }
 
     xprompt () { # Usage xprompt [MSG] [BG_COLOR]
-	printf '%s' "" | ${FM_PROG} ${FM_OPT_REVERSE} ${FM_OPT_PROMPT} "SUDO: $1 "
+	printf '%s' "" | ${FM_PROG} $(makePromptOpts reverse prompt) "SUDO: $1 "
     }
 
     DangerPrompt () { # Usage: DangerPrompt [MSG] && ...
-	[ "$(printf "No\\nYes" | ${FM_PROG} ${FM_OPT_REVERSE} ${FM_OPT_PROMPT} "SUDO: $1 ")" = "Yes" ]
+	[ "$(printf "No\\nYes" | ${FM_PROG} $(makePromptOpts reverse prompt) "SUDO: $1 ")" = "Yes" ]
```

Comment:
I suppose, the main reason I came on this approach is because it is a bit more flexible for allowing different types of prompts. As can be seen in the script, some of the prompts have the no-sort option set, while others don't. So, in order to have that split we would have to have it separate like this anyways. Example:
We have variable for options, lets call it `FM_OPTS`. If `FM_OPTS` is equal to `--reverse --no-sort --prompt` then we must use the no-sort flag on every prompt with no way out to take it out, unless we make a second variable for all opts minus `--no-sort`, which would be more redundant.

What could work well is doing this but with an extra check (your suggestion) to see if the user has defined one of these flags themselves like was done elsewhere. i.e

```shell
"bemenu")
		[ -z "$FM_OPT_PROMPT" ] && FM_OPT_PROMPT="-p"
		;;
```

Reply:
To be honest, I don't consider `--no-sort` option in `fzf` is necessary because `dmenu` have no such option, and the reason I use `--no-sort` is that it break the order of presenting dir/files: `Dirs -> Files -> DotDirs -> DotFiles`. I want to give user the freedom to use this file manager on `fzf` without this designated order.

The way we design `dmenufm` really makes only prompt is important. Like dmenu have color options but clearly I didn't specify color option in this `fzf` branch, and it works just like `dmenu` without specifying color. 

I think the test for options is really nice.

Do you think it is a good idea for us to provide support for most of the menu/fuzzy finder system, or we use a caveat that only fully support `dmenu` and not for other menu system?

Reply:
I'd say focus should be on dmenufm, as that is what it is designed to work with but I am interested in making it more accessible out of the box to similar programs they like to use.

I suppose from your reasoning we could have a more global user options variable for variables used on every prompt instead of many individual ones. I'll work on this some more today/tomorrow and push some changes to see how it goes. Basic idea now is to use case statement to select the `--prompt` equivalent for the program, and then have a global variables option.

Reply:
Thx! Excited for your work!

Now I am preparing for PhD qualifier during quarantine...my life is just miserable. I cannot guarantee how much I can help, but I will do my best conditional on preparing qualifier.

Reply:
Ah alright, best of luck to you! Tough times right now so take it easy. If you ever need help/want with this project I'll get around to it and you can contact me here or via email.